mantidproject/mantid
# Faster, simpler EventList::sortTof

Description of work.

This replaces a lot of custom logic with `tbb::parallel_for` and `tbb::parallel_sort`. Besides being simpler code ( 324 fewer lines!), it also appears to be significantly faster. I ran `DataObjectsTest.EventListTestPerformance` and the total time decreased from 2.03 seconds to 1.22 seconds. `test_sort_tof`decreased from 0.29 seconds to 0.13 seconds. 

**To test:**



Try this on larger, "real" data and verify that performance still improves.

There is no GitHub issue associated with this pull request.

Release notes?

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
